modules/lofreq/1.0/lofreq.smk

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Bruno Grande
+# Module Author:    Bruno Grande
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["lofreq"]`
+CFG = op.setup_module(
+    name = "lofreq",
+    version = "1.0",
+    subdirectories = ["inputs", "lofreq", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _lofreq_input_bam,
+    _lofreq_output_vcf,
+    _lofreq_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _lofreq_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+
+# Run LoFreq in somatic variant calling mode
+rule _lofreq_run:
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
+        normal_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        dbsnp = reference_files("genomes/{genome_build}/variation/dbsnp.common_all-151.vcf.gz")
+    output:
+        vcf_snvs = CFG["dirs"]["lofreq"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/somatic_final_minus-dbsnp.snvs.vcf.gz",
+        vcf_indels = CFG["dirs"]["lofreq"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/somatic_final_minus-dbsnp.indels.vcf.gz"
+    log:
+        stdout = CFG["logs"]["lofreq"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/lofreq.stdout.log",
+        stderr = CFG["logs"]["lofreq"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/lofreq.stderr.log"
+    params:
+        opts = CFG["options"]["lofreq"],
+        regions = op.switch_on_wildcard("seq_type", CFG["switches"]["regions_bed"])
+    conda:
+        CFG["conda_envs"]["lofreq"]
+    threads:
+        CFG["threads"]["lofreq"]
+    resources:
+        mem_mb = op.retry(CFG["mem_mb"]["lofreq"])
+    shell:
+        op.as_one_line("""
+        lofreq somatic {params.opts} --threads {threads} -t {input.tumour_bam} -n {input.normal_bam}
+        -f {input.fasta} -o $(dirname {output.vcf_snvs})/ -d {input.dbsnp} {params.regions} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
```

Comment:
Should we incorporate vcf2maf into this (and all variant callers) directly? It's not clear to me if that will be optional and, if so, how.

Reply:
Are you thinking about incorporating the whole module here? I think that might not be necessary, as we can combine them together in the project snakefile - similar to what is in demo project. Alternatively, I can do an if statement based on config, and that if will also include all rule - so the final output will be the maf file.

Reply:
Since Ryan made this comment before he'd made a vcf2maf module, I'd say this is no longer relevant.

Reply:
Agree with Laura. Obsolete comment